recipes/openvino/all/conanfile.py

```diff
@@ -0,0 +1,381 @@
+        # misc
+        if self._preprocessing_available:
+            toolchain.cache_variables["ENABLE_GAPI_PREPROCESSING"] = True
+        toolchain.cache_variables["BUILD_SHARED_LIBS"] = self.options.shared
+        toolchain.cache_variables["CPACK_GENERATOR"] = "CONAN"
+        toolchain.cache_variables["ENABLE_PROFILING_ITT"] = False
+        toolchain.cache_variables["ENABLE_PYTHON"] = False
+        toolchain.cache_variables["ENABLE_PROXY"] = False
+        toolchain.cache_variables["ENABLE_WHEEL"] = False
+        toolchain.cache_variables["ENABLE_CPPLINT"] = False
+        toolchain.cache_variables["ENABLE_NCC_STYLE"] = False
+        toolchain.cache_variables["ENABLE_SAMPLES"] = False
+        toolchain.cache_variables["ENABLE_TEMPLATE"] = False
+        toolchain.generate()
+
+    def validate_build(self):
+        if self.settings.compiler.get_safe("cppstd"):
+            check_min_cppstd(self, "11")
+
+        minimum_version = self._compilers_minimum_version.get(str(self.settings.compiler), False)
+        compiler_version = Version(self.settings.compiler.version)
+        if minimum_version and compiler_version < minimum_version:
+            raise ConanInvalidConfiguration(
+                f"{self.ref} requires {self.settings.compiler} ver. {minimum_version}, provided ver. {compiler_version}.",
+            )
+
+        # OpenVINO has unresolved symbols, when clang is used with libc++
+        if self.settings.compiler == "clang" and self.settings.compiler.libcxx == "libc++":
+            raise ConanInvalidConfiguration(
+                f"{self.ref} cannot be built with clang and libc++ due to unresolved symbols. "
+                f"Please, use libstdc++ instead."
+            )
+
+        if self.settings.os == "Emscripten":
+            raise ConanInvalidConfiguration(f"{self.ref} does not support Emscripten")
+
+        # TODO: resolve it later, since it is not critical for now
+        # Conan Center CI fails with our of memory error when building OpenVINO
+        if self.settings.build_type == "Debug":
+            raise ConanInvalidConfiguration(f"{self.ref} does not support Debug build type")
+
+    def validate(self):
+        # dynamically compiled frontends cannot be run against dynamic protobuf
+        if self._protobuf_required and self.options.shared and self.dependencies["protobuf"].options.shared:
+            raise ConanInvalidConfiguration(f"{self.ref}:shared=True requires protobuf:shared=False for correct work.")
+
+        if self.options.get_safe("enable_gpu"):
+            # GPU does not support oneDNN in static build configuration, warn about it
+            if not self.options.shared:
+                self.output.warning(f"{self.name} recipe builds GPU plugin without oneDNN (dGPU) support during static build.")
+
+            # GPU plugin is currently not stable when compiled with clang compiler
+            if self.settings.compiler == "clang":
```

Comment:
When say `clang` does it include Clang from MacOS too?

Reply:
OpenVINO GPU plugin is not available on macOS